aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,198 @@
+      Optional<Table> glueTableOptional = getGlueTable();
+      if (glueTableOptional.isPresent()) {
+        glueTable = glueTableOptional.get();
+        isUpdate = true;
+        // If we try to create the table but the metadata location is already set, then we had a concurrent commit
+        if (base == null && glueTable.parameters().get(METADATA_LOCATION_PROP) != null) {
+          throw new AlreadyExistsException("Cannot commit because table %s already exists in Glue", tableName());
+        }
+      }
+
+      checkMetadataLocation(isUpdate, glueTable, base);
+      Map<String, String> properties = prepareProperties(isUpdate, glueTable, newMetadataLocation);
+      persistGlueTable(isUpdate, properties);
+      exceptionThrown = false;
+    } catch (CommitFailedException | AlreadyExistsException e) {
+      throw e;
+    } catch (ConcurrentModificationException e) {
+      throw new CommitFailedException(e,
+          "Cannot commit %s because Glue detected concurrent update", tableName());
+    } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
+      throw new AlreadyExistsException(e,
+          "Cannot commit %s because its Glue table already exists when trying to create one", tableName());
+    } catch (SdkException e) {
+      throw new CommitFailedException(e, "Cannot commit %s because unexpected exception contacting AWS", tableName());
+    } finally {
+      if (exceptionThrown) {
+        io().deleteFile(newMetadataLocation);
+      }
+    }
+  }
+
+  private void checkMetadataLocation(boolean isUpdate, Table glueTable, TableMetadata base) {
+    String glueMetadataLocation = isUpdate ? glueTable.parameters().get(METADATA_LOCATION_PROP) : null;
+    String baseMetadataLocation = base != null ? base.metadataFileLocation() : null;
+    if (!Objects.equals(baseMetadataLocation, glueMetadataLocation)) {
+      throw new CommitFailedException(
+          "Cannot commit %s because base metadata location '%s' is not same as the current Glue location '%s'",
+          tableName(), baseMetadataLocation, glueMetadataLocation);
+    }
+  }
+
+  private Optional<Table> getGlueTable() {
+    try {
+      GetTableResponse response = glue.getTable(GetTableRequest.builder()
+          .catalogId(awsProperties.glueCatalogId())
+          .databaseName(databaseName)
+          .name(tableName)
+          .build());
+      return Optional.ofNullable(response.table());
+    } catch (EntityNotFoundException e) {
+      return Optional.empty();
+    }
+  }
+
+  private Map<String, String> prepareProperties(boolean isUpdate, Table glueTable, String newMetadataLocation) {
+    Map<String, String> properties = isUpdate ? Maps.newHashMap(glueTable.parameters()) : Maps.newHashMap();
+    properties.put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toUpperCase(Locale.ENGLISH));
+    properties.put(METADATA_LOCATION_PROP, newMetadataLocation);
+    if (currentMetadataLocation() != null && !currentMetadataLocation().isEmpty()) {
```

Comment:
If `currentMetadataLocation` is an empty string, then there is a problem. Empty strings shouldn't be handles like null strings.

Reply:
That looks wrong to me, but if that's the currently released behavior then let's keep it. :frowning_face: